src/viur/core/bones/phone.py

```diff
@@ -0,0 +1,129 @@
+import re
+import typing as t
+import json
+
+from viur.core.bones.string import StringBone
+from viur.core.bones.base import ReadFromClientError, ReadFromClientErrorSeverity
+from viur.core import conf
+
+from pathlib import Path
+
+DEFAULT_REGEX = r"^\+?(\d{1,3})[-\s]?(\d{1,4})[-\s]?(\d{1,4})[-\s]?(\d{1,9})$"
+
+COUNTRYFILE = Path(__file__).parent / "../languages/country_information.json"
+
+with open(COUNTRYFILE, "r") as fin:
+    COUNTRYDATA = json.load(fin)
+
+
+class PhoneBone(StringBone):
+    """
+    The PhoneBone class is designed to store validated phone/fax numbers in configurable formats.
+    This class provides a number validation method, ensuring that the given phone/fax number conforms to the
+    required/configured format and structure.
+    """
+
+    type: str = "str.phone"
+    """
+    A string representing the type of the bone, in this case "str.phone".
+    """
+
+    def __init__(
+        self,
+        *,
+        test: t.Optional[t.Pattern[str]] = DEFAULT_REGEX,
+        max_length: int = 15,  # maximum allowed numbers according to ITU-T E.164
+        default_country_code: t.Optional[str] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        """
+        Initializes the PhoneBone with an optional custom regex for phone number validation, a default country code,
+        and a flag to apply the default country code if none is provided.
+
+        :param test: An optional custom regex pattern for phone number validation.
+        :param max_length: The maximum length of the phone number. Passed to 'StringBone'.
+        :param default_country_code: The default country code to apply if none is provided.
```

Comment:
I find this none here unfavourable. Does it mean &laquo;_if `None` is provided_&raquo;   or  &laquo;_if no country code is provided in the client data_&raquo;?

Reply:
This is for the bone definition, if u apply it in the definition the default country code will be prefixed automatically. if no default_country_code is given in the definition of the Bone, there will be no automatic appliance of any country code. But if a client adds 0049 for example the 00 will be replaced by a +. It was wanted like this